src/gui/shareusergroupwidget.cpp

```diff
@@ -703,59 +706,58 @@ void ShareUserLine::slotEditPermissionsChanged()
     // Can never manually be set to "partial".
     // This works because the state cycle for clicking is
     // unchecked -> partial -> checked -> unchecked.
-    if (_ui->permissionsEdit->checkState() == Qt::PartiallyChecked) {
-        _ui->permissionsEdit->setCheckState(Qt::Checked);
-    }
+//    if (_ui->permissionsEdit->checkState() == Qt::PartiallyChecked) {
+//        _ui->permissionsEdit->setCheckState(Qt::Checked);
+//    }
 
     Share::Permissions permissions = SharePermissionRead;
 
     //  folders edit = CREATE, READ, UPDATE, DELETE
     //  files edit = READ + UPDATE
-    if (_ui->permissionsEdit->checkState() == Qt::Checked) {
 
         /*
          * Files can't have create or delete permisisons
          */
         if (!_isFile) {
-            if (_permissionChange->isEnabled())
+            if (_permissionChange->isEnabled()){
                 permissions |= SharePermissionUpdate;
-            if (_permissionCreate->isEnabled())
+               _ui->currentPermission->setText("Can Edit");
+            }
+            /*if (_permissionCreate->isEnabled())
                 permissions |= SharePermissionCreate;
             if (_permissionDelete->isEnabled())
-                permissions |= SharePermissionDelete;
+                permissions |= SharePermissionDelete;*/
         } else {
             permissions |= SharePermissionUpdate;
+
         }
-    }
 
     if(_isFile && _permissionReshare->isEnabled() && _permissionReshare->isChecked())
         permissions |= SharePermissionShare;
 
     _share->setPermissions(permissions);
 }
 
-void ShareUserLine::slotPermissionsChanged()
-{
+void ShareUserLine::slotPermissionsChanged(){
     setEnabled(false);
 
     Share::Permissions permissions = SharePermissionRead;
 
-    if (_permissionReshare->isChecked())
+    if (_permissionReshare->isChecked()){
         permissions |= SharePermissionShare;
-
-    if (!_isFile) {
-        if (_permissionChange->isChecked())
-            permissions |= SharePermissionUpdate;
-        if (_permissionCreate->isChecked())
-            permissions |= SharePermissionCreate;
-        if (_permissionDelete->isChecked())
-            permissions |= SharePermissionDelete;
-    } else {
-        if (_ui->permissionsEdit->isChecked())
-            permissions |= SharePermissionUpdate;
+         _share->setPermissions(permissions);
     }
-
-    _share->setPermissions(permissions);
+    else if(_permissionRead->isChecked()){
+        permissions |= SharePermissionRead;
+        _ui->currentPermission->setText("Read Only");
+         _share->setPermissions(permissions);
+    }
+    else if(_permissionChange->isChecked()){
+        permissions |= SharePermissionUpdate;
+        _ui->currentPermission->setText("Can Edit");
+         _share->setPermissions(permissions);
+    }
+   // _share->setPermissions(permissions);
```

Comment:
1. Please delete unneeded code instead of commenting out
2. For multiple if-else statements please do:

if (x) {

} else {

}